certgen/certProcessor/src/main/java/org/incredible/certProcessor/views/HTMLZipProcessor.java

```diff
@@ -0,0 +1,102 @@
+package org.incredible.certProcessor.views;
+
+import org.apache.commons.io.IOUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.*;
+import java.net.URL;
+import java.util.Enumeration;
+import java.util.zip.ZipEntry;
+import java.util.zip.ZipFile;
+
+public class HTMLZipProcessor extends HTMLTemplateProvider {
+
+
+    private String content = null;
+
+    private static Logger logger = LoggerFactory.getLogger(HTMLZipProcessor.class);
+
+    private URL zipUrl;
+
+    public HTMLZipProcessor(URL zipUrl) {
+        this.zipUrl = zipUrl;
+    }
+
+    private static final int bufferSize = 4096;
+
+    private void getZipFileFromURl(File targetDirectory) throws IOException {
+        if (!targetDirectory.exists()) {
+            targetDirectory.mkdirs();
+        }
+        InputStream inputStream = new BufferedInputStream(zipUrl.openStream());
+        // make sure we get the actual file
+        File zipFile = File.createTempFile("arc", ".zip", targetDirectory);
+        OutputStream outputStream = new BufferedOutputStream(new FileOutputStream(zipFile));
+        extractFile(inputStream, outputStream);
+        outputStream.close();
+        unzip(zipFile, targetDirectory);
+    }
+
+    private void extractFile(InputStream inputStream, OutputStream outputStream) throws IOException {
+        byte[] buffer = new byte[bufferSize];
+        int length = inputStream.read(buffer);
+        while (length >= 0) {
+            outputStream.write(buffer, 0, length);
+            length = inputStream.read(buffer);
+        }
+        inputStream.close();
+        outputStream.close();
+    }
+
+    private void unzip(File zip, File targetDirectory) throws IOException {
+        if (!zip.exists())
+            throw new IOException(zip.getAbsolutePath() + " does not exist");
+        if (!isDirectoryExists(targetDirectory))
+            throw new IOException("Could not create directory: " + targetDirectory);
+        ZipFile zipFile = new ZipFile(zip);
+        for (Enumeration entries = zipFile.entries(); entries.hasMoreElements(); ) {
+            ZipEntry entry = (ZipEntry) entries.nextElement();
+            File file = new File(targetDirectory, File.separator + entry.getName());
+            if (!isDirectoryExists(file.getParentFile()))
+                throw new IOException("Could not create directory: " + file.getParentFile());
+            if (!entry.isDirectory()) {
+                extractFile(zipFile.getInputStream(entry), new BufferedOutputStream(new FileOutputStream(file)));
+                if (entry.getName().endsWith(".html")) {
```

Comment:
Include a check to assert there is only 1 file with .html extension.
And request to add javadocs in all these functions.